onnxruntime/core/providers/nnapi/nnapi_builtin/nnapi_lib/NeuralNetworksWrapper.cc

```diff
@@ -100,7 +101,7 @@ size_t OperandType::GetElementByteSize() const {
 
 size_t OperandType::GetOperandBlobByteSize() const {
   size_t num_elements = std::accumulate(dimensions.begin(), dimensions.end(), 1, std::multiplies<size_t>());
-  return num_elements * GetElementByteSize();
+  return SafeInt<size_t>(num_elements * GetElementByteSize());
 }
```

Comment:
I think you need to convert one to a SafeInt prior to doing the calculation. Otherwise you're putting the result of the calculation in the SafeInt and any overflow has already happened.
e.g. SafeInt<size_t>(num_elements) * GetElementByteSize();

Reply:
But isn't num_elements already a size_t?

Reply:
The SafeInt will check if the size_t overflows when num_elements is multiplied by GetElementByteSize(). more likely on a 32-bit system where size_t is smaller. 

---
In reply to: [569776202](https://github.com/microsoft/onnxruntime/pull/6510#discussion_r569776202) [](ancestors = 569776202)